src/Application.vala

```diff
@@ -307,6 +318,16 @@ public class AppCenter.App : Gtk.Application {
 
         return msg;
     }
+
+    private static string? mimetype_from_file (File file) {
+        string uri = file.get_uri ();
+        string[] tokens = uri.split (Path.DIR_SEPARATOR_S);
+        if (tokens.length < 2) {
+            return null;
+        }
+
+        return "%s/%s".printf (tokens[tokens.length - 2], tokens[tokens.length - 1]);
```

Comment:
You should just be able to use negative indices here i.e. tokens[-2], tokens[-1].

Reply:
This doesn't work for me, after I change it to use negative indices, the process segfaults.

Reply:
Yes, sorry - I was thinking of string slicing indices.  It doesn't work in Vala as an actual index.